openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java

```diff
@@ -141,4 +148,32 @@ public static String getValidName(String identifier) {
         }
         return identifier.substring(0, 1).toLowerCase(Locale.ENGLISH) + identifier.substring(1);
     }
+
+    /**
+     * This util function uses to take the field value from annotation field.
+     *
+     * @param annotations           - Annotation node list
+     * @param annotationReference   - Annotation reference that needs to extract
+     * @param annotationField       - Annotation field name that uses to take value
+     * @return                      - string value of field
+     */
+    public static String extractServiceAnnotationDetails(NodeList<AnnotationNode> annotations,
+                                                   String annotationReference, String annotationField) {
+        String mediaType = "";
+        for (AnnotationNode annotation: annotations) {
+            Node annotReference = annotation.annotReference();
+            if (annotReference.toString().trim().equals(annotationReference) && annotation.annotValue().isPresent()) {
+                MappingConstructorExpressionNode listOfannotValue = annotation.annotValue().get();
+                for (MappingFieldNode field : listOfannotValue.fields()) {
+                    SpecificFieldNode media = (SpecificFieldNode) field;
+                    if ((media).fieldName().toString().trim().equals(annotationField)
+                            && media.valueExpr().isPresent()) {
+                        ExpressionNode expressionNode = media.valueExpr().get();
+                        mediaType = expressionNode.toString().trim().replaceAll("\"", "");
+                    }
+                }
+            }
+        }
+        return mediaType;
+    }
```

Comment:
Shall we use the `Optional` APIs so that the util will correctly indicates whether any values found for the given annotation reference + field?

Reply:
Sure It will be more clear about the return value availability. will change this way.